node-js-web-server/server220.js

```diff
@@ -0,0 +1,38 @@
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
+
+http.createServer(function (request, response) {
+   console.log('request url: ', request.url);
+
+   var filePath = './public' + request.url;
+   if (filePath == './public/') {
+      filePath = './public/index.html';
+   }
+   console.log(filePath)
+
+   var extName = String(path.extname(filePath)).toLowerCase();
+   var mimeTypes = {
+      '.html': 'text/html',
+      '.js': 'text/javascript',
+      '.css': 'text/css',
+      '.png': 'image/png'
+   };
+   var contentType = mimeTypes[extName] || 'application/octet-stream';
+
+   fs.readFile(filePath, function(error, content) {
+      if (error) {
+         fs.readFile('./public/404.html', function(error, content) {
+            
+        }); 
+      } else {
+         
+      }
+   });
+}).listen(8080);
+console.log('Server is running at http://localhost:8080');
+
+// now it is time for you to do something with the `response` object.
```

Comment:
Now it's time for you to do something with the `response` object.
To show the user if there's a 404 error, call the `writeHead` method from `response`.